crates/pathfinder/src/state/sync.rs

```diff
@@ -207,6 +207,11 @@ where
                         SyncStatus::Status(status) => {
                             status.current_block_hash = block_hash;
                             status.current_block_num = StarknetBlockNumber(block_num);
+
+                            if status.highest_block_num.0 <= block_num {
+                                status.highest_block_num = StarknetBlockNumber(block_num);
+                                status.highest_block_hash = block_hash;
+                            }
```

Comment:
Feels like a missed opportunity for the Ord:

```suggestion
                            status.highest = status.highest.max(status.current);
```

But I don't know how to do the serde tricks this requires..

Reply:
Yeah I think I would make a new struct that isn't the weird serde json one so its more sensible.

Reply:
There's a way to do this with `serde_with::with_prefix!`, I can follow up with that.